lib/signaling/v2/localparticipant.js

```diff
@@ -53,6 +75,21 @@ class LocalParticipantV2 extends LocalParticipantSignaling {
     });
   }
 
+  /**
+   * Update the {@link BandwidthProfileOptions}.
+   * @param {BandwidthProfileOptions} bandwidthProfile
+   */
+  setBandwidthProfile(bandwidthProfile) {
+    if (!isDeepEqual(this._bandwidthProfile, bandwidthProfile)) {
+      // NOTE(mmalavalli): Object.assign() copies the values of only
+      // the top level properties. In order to deep copy the object, we
+      // stringify and parse the object.
+      this._bandwidthProfile = JSON.parse(JSON.stringify(bandwidthProfile));
```

Comment:
@makarandp0 I want your opinion on the way I perform a deep copy of the object here. I'm not to sure about it, so any thoughts you may have will be helpful here.

Reply:
this looks fine, as it will get rid of any non-value types from the bandwidthProfile. 
This code won't work if  `bandwidthProfile = undefined`. But I guess that won't happen because validateBandwidthProfile would reject it. I suggest add a comment saying why  `bandwidthProfile = undefined` is not possible here.  Or even better add an assert 
```
assert(bandwidthProfile !== undefined, 'something went wrong undefined should not be here`);
```